### PR TITLE
Provide information about the interactive session

### LINKED ISSUE #31: Provide Session Info
# Description

The JupyterLab extension should provide a feature for getting information about that session. That page should include a link to return to the project into Renku.

This is the second feature in https://github.com/SwissDataScienceCenter/renku-ui/issues/1155.

### PR DESCRIPTION
**Pull request by ciyer**
Fix #31 

To test, you can launch a session in the jl-extension project on dev: https://dev.renku.ch/projects/cramakri/jl-extension/environments/new 

Use the jl-extension branch to get the most current version.

Todo:
- [x] Refresh session info on branch checkout or HEAD move

### CODE REVIEW

**Comment by ciyer** [comment]
Currently experiencing the problem documented here https://github.com/krassowski/jupyterlab-lsp/issues/375#issuecomment-761078622 Need to try the same solution.

File: jl_renku/core.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from git import Repo
+
+
+def renkulab_url_from_gitlab_url(gitlab_url):
```
Comment (rokroskar) [inline jl_renku/core.py:21]: You might find this class useful - it's robust for all kinds of different ways that one can specify a repo URL: https://github.com/SwissDataScienceCenter/renku-python/blob/0cf51f64fe29be415980199e240fd04c3edfb604/renku/core/models/git.py#L82-L83

Comment (rokroskar) [reply jl_renku/core.py:21]: You can see all the cases it covers here: https://github.com/SwissDataScienceCenter/renku-python/blob/a2e3297a318dc07c611d45eda059e73e67dd152a/tests/core/models/test_git.py#L27

Comment (ciyer) [reply jl_renku/core.py:21]: That is good to know. I think the implementation that is here should be thought of as a stub to get the interactions working. I think we would switch to using the renku-core-service before we release this. Or, we might drop the plug-in completely if https://github.com/SwissDataScienceCenter/renku/issues/1907 works out.

**Review by vfried — CHANGES_REQUESTED** [review]
I tested the widget and it works very well.

I still have some details i think could be changed.

1) Could we align this, i see commit is a bit more on the left.
image

2) We could add here a small message that says "run git status to see changes"
image

3) Do we need this file?
image

3) There are a few more things i would like to see here, this doesn't have to be part of the current PR, we could do this later:
- A  message that says if you can commit to the project or not.
- A message if something was committed but needs to be pushed (right now you only see a message if something needs to be committed).

**Comment by ciyer** [comment]
Thanks for these suggestions! 

1 and 2 have been fixed. The UI now looks like: 

image

3. This is part of the official structure of an extension, but I refactored so that this file is correctly being used.

4. Good ideas, but I would prefer to address in a later PR if we decide to continue with the extension.

**Comment by vfried** [comment]
Thanks for the fixes! I agree with point 4, we should wait and see where the extension goes. 

Regarding point 2 I see in the image that things are still misaligned on the left. Is this intentional?

**Comment by ciyer** [comment]
> Thanks for the fixes! I agree with point 4, we should wait and see where the extension goes.
> 
> Regarding point 2 I see in the image that things are still misaligned on the left. Is this intentional?

The column has to be left, center, or right aligned. I think it looks best right-justified, with ragged left.

**Comment by ciyer** [comment]
Oh, and the "view in RenkuLab" button isn't aligned with the "git status" info, but this is caused by the interaction between the Jupyter styles that are being used in the button/link and our styles. I wasn't able to find a quick fix, so I would prefer to leave it as is for the moment.

**Review by vfried — APPROVED** [review]
I understand using jupyterlab's styles is a bit limited but has its benefits.
Thanks for the fixes!

**Pull request merged**